contracts/NftFactory.sol

```diff
@@ -104,4 +119,20 @@ contract NftFactory is UUPSUpgradableTemplate {
     (bool success, ) = beneficiary.call{value: amount}("");
     require(success);
   }
+
+  function buyTokensWithSeeds(uint8 nftId, uint256 amount) external {
+    require(msg.value >= _pricesInSeed[nftId].mul(amount), "NftFactory: insufficient payment");
+    seedProceedsBalance += msg.value;
+    seedToken.transferFrom(_msgSender(), address(this), _pricesInSeed[nftId].mul(amount));
```

Comment:
After paying, we must mint the NFT for the user. Look at last line in `buyTokens`

Reply:
Oh right, I deleted that line by mistake

Reply:
found another error: in withdrawSeedProceeds, I think this line:
```solidity
    (bool success, ) = beneficiary.call{value: amount}("");
```
should be:
```solidity
    seedToken.transferFrom(address(this), beneficiary, _pricesInSeed[nftId].mul(amount));
```

Reply:
Yes, I didn't look at that function because I was driving. The original call is to transfer native tokens only.